### PR TITLE
update go version to 1.22.4

### PR DESCRIPTION
**Pull request by dhiaayachi**
### Description
upgrade go version to 1.22.4

### Testing & Reproduction steps

- automated tests

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [ ] appropriate backport labels added
* [ ] not a security concern

### CODE REVIEW

File: .changelog/21265.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:enhancement
```
Comment (DanStough) [inline .changelog/21265.txt:1]: Sometime we label this as `security` and list the CVE fixed. Do we know if this affects `consul`?

**Review by DanStough — APPROVED** [review]
Is this something we intend to backport into 1.19.0? FYI that might have to be manual.

File: .changelog/21265.txt (hunk 1 above)
Comment (dhiaayachi) [reply .changelog/21265.txt:1]: I don't think it affect consul, I tried runing govuln on Consul with older versions and it didn't show up.

**Comment by dhiaayachi** [comment]
> Is this something we intend to backport into 1.19.0? FYI that might have to be manual.

@DanStough This will be backported to all the supported versions, I added the all label and we will see if any manual intervention will be needed.

**Comment by hc-github-team-consul-core** [comment]
@dhiaayachi, a backport is missing for this PR [21265] for versions [1.16] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Pull request merged**